include/SetupDialog.h

```diff
@@ -67,148 +67,136 @@ protected slots:
+	void toggleDisableBackup(bool enabled);
+	void toggleOpenLastProject(bool enabled);
+	void setLanguage(int lang);
+
+	// Performance settings widget.
+	void setAutoSaveInterval(int time);
 	void resetAutoSave();
+	void toggleAutoSave(bool enabled);
+	void toggleRunningAutoSave(bool enabled);
+	void toggleSmoothScroll(bool enabled);
+	void toggleAnimateAFP(bool enabled);
+	void toggleSyncVSTPlugins(bool enabled);
+	void vstEmbedMethodChanged();
+	void toggleVSTAlwaysOnTop(bool en);
+	void toggleDisableAutoQuit(bool enabled);
 
-	// audio settings widget
-	void audioInterfaceChanged( const QString & _driver );
-
-	// MIDI settings widget
-	void midiInterfaceChanged( const QString & _driver );
-
+	// Audio settings widget.
+	void audioInterfaceChanged(const QString & driver);
+	void toggleHQAudioDev(bool enabled);
+	void setBufferSize(int value);
+	void resetBufferSize();
 
-	void toggleToolTips( bool _enabled );
-	void toggleWarnAfterSetup( bool _enabled );
-	void toggleDisplaydBFS( bool _enabled );
-	void toggleMMPZ( bool _enabled );
-	void toggleDisableBackup( bool _enabled );
-	void toggleOpenLastProject( bool _enabled );
-	void toggleHQAudioDev( bool _enabled );
+	// MIDI settings widget.
+	void midiInterfaceChanged(const QString & driver);
 
+	// Paths settings widget.
 	void openWorkingDir();
+	void setWorkingDir(const QString & workingDir);
 	void openVSTDir();
-	void openGIGDir();
-	void openSF2Dir();
-	void openArtworkDir();
+	void setVSTDir(const QString & vstDir);
 	void openLADSPADir();
-	void openSTKDir();
-	void openDefaultSoundfont();
-	void openBackgroundArtwork();
-
-	void toggleSmoothScroll( bool _enabled );
-	void toggleAutoSave( bool _enabled );
-	void toggleRunningAutoSave( bool _enabled );
-	void toggleOneInstrumentTrackWindow( bool _enabled );
-	void toggleCompactTrackButtons( bool _enabled );
-	void toggleSyncVSTPlugins( bool _enabled );
-	void toggleAnimateAFP( bool _enabled );
-	void toggleNoteLabels( bool en );
-	void toggleDisplayWaveform( bool en );
-	void toggleDisableAutoquit( bool en );
-
-	void vstEmbedMethodChanged();
-	void toggleVSTAlwaysOnTop( bool en );
-
-	void setLanguage( int lang );
+	void setLADSPADir(const QString & ladspaDir);
+	void openSF2Dir();
+	void setSF2Dir(const QString & sf2Dir);
+	void openSF2File();
+	void setSF2File(const QString & sf2File);
+	void openGIGDir();
+	void setGIGDir(const QString & gigDir);
+	void openThemeDir();
+	void setThemeDir(const QString & themeDir);
+	void openBackgroundPicFile();
+	void setBackgroundPicFile(const QString & backgroundPicFile);
 
+	void showRestartWarning();
 
 private:
 	TabBar * m_tabBar;
 
-	QSlider * m_bufSizeSlider;
-	QLabel * m_bufSizeLbl;
-	int m_bufferSize;
-
-	bool m_toolTips;
-	bool m_warnAfterSetup;
+	// General settings widgets.
 	bool m_displaydBFS;
+	bool m_tooltips;
+	bool m_displayWaveform;
+	bool m_printNoteLabels;
+	bool m_compactTrackButtons;
+	bool m_oneInstrumentTrackWindow;
 	bool m_MMPZ;
 	bool m_disableBackup;
```

Comment:
Maybe out of scope, but the name of this variable should be `m_enableBackup`.

Reply:
The name is misleading, but I think this should go into a separate pull request.